cider/tests/functionality/CiderFilterTest.cpp

```diff
@@ -20,265 +20,157 @@
-        QueryDataGenerator::generateBatchByTypes(99,
-                                                 {"col_1", "col_2", "col_3", "col_4"},
-                                                 {CREATE_SUBSTRAIT_TYPE(I32),
-                                                  CREATE_SUBSTRAIT_TYPE(I64),
-                                                  CREATE_SUBSTRAIT_TYPE(Fp32),
-                                                  CREATE_SUBSTRAIT_TYPE(Fp64)},
-                                                 {7, 7, 7, 7}))};
+        *(new CiderBatch(schema, array, std::make_shared<CiderDefaultAllocator>())))};
   }
 };
 
 class CiderProjectAllTestBase : public CiderTestBase {
  public:
   CiderProjectAllTestBase() {
     table_name_ = "test";
+    // TODO(yizhong): Enable this after date and bool is supported in arrow.
+    // create_ddl_ =
+    //     "CREATE TABLE test(col_1 INTEGER, col_2 BIGINT, col_3 TINYINT, col_4 SMALLINT,
+    //     " "col_5 FLOAT, col_6 DOUBLE, col_7 DATE, col_8 BOOLEAN);";
+    // input_ = {std::make_shared<CiderBatch>(QueryDataGenerator::generateBatchByTypes(
+    //     100,
+    //     {"col_1", "col_2", "col_3", "col_4", "col_5", "col_6", "col_7", "col_8"},
+    //     {CREATE_SUBSTRAIT_TYPE(I32),
+    //      CREATE_SUBSTRAIT_TYPE(I64),
+    //      CREATE_SUBSTRAIT_TYPE(I8),
+    //      CREATE_SUBSTRAIT_TYPE(I16),
+    //      CREATE_SUBSTRAIT_TYPE(Fp32),
+    //      CREATE_SUBSTRAIT_TYPE(Fp64),
+    //      CREATE_SUBSTRAIT_TYPE(Date),
+    //      CREATE_SUBSTRAIT_TYPE(Bool)},
+    //     {1, 2, 2, 2, 3, 3, 4, 2},
+    //     GeneratePattern::Random))};
+
     create_ddl_ =
-        "CREATE TABLE test(col_1 INTEGER, col_2 BIGINT, col_3 TINYINT, col_4 SMALLINT, "
-        "col_5 FLOAT, col_6 DOUBLE, col_7 DATE, col_8 BOOLEAN);";
-    input_ = {std::make_shared<CiderBatch>(QueryDataGenerator::generateBatchByTypes(
+        R"(CREATE TABLE test(col_1 INTEGER, col_2 BIGINT, col_3 TINYINT, col_4 SMALLINT, col_5 FLOAT, col_6 DOUBLE);)";
+    QueryArrowDataGenerator::generateBatchByTypes(
+        schema,
+        array,
         100,
-        {"col_1", "col_2", "col_3", "col_4", "col_5", "col_6", "col_7", "col_8"},
+        {"col_1", "col_2", "col_3", "col_4", "col_5", "col_6"},
         {CREATE_SUBSTRAIT_TYPE(I32),
          CREATE_SUBSTRAIT_TYPE(I64),
          CREATE_SUBSTRAIT_TYPE(I8),
          CREATE_SUBSTRAIT_TYPE(I16),
          CREATE_SUBSTRAIT_TYPE(Fp32),
-         CREATE_SUBSTRAIT_TYPE(Fp64),
-         CREATE_SUBSTRAIT_TYPE(Date),
-         CREATE_SUBSTRAIT_TYPE(Bool)},
-        {1, 2, 2, 2, 3, 3, 4, 2},
-        GeneratePattern::Random))};
+         CREATE_SUBSTRAIT_TYPE(Fp64)},
+        {1, 2, 2, 2, 3, 3},
+        GeneratePattern::Random);
+    input_ = {std::make_shared<CiderBatch>(
+        *(new CiderBatch(schema, array, std::make_shared<CiderDefaultAllocator>())))};
   }
 };
 
 TEST_F(CiderProjectAllTestBase, filterProjectAllTest) {
-  assertQuery("SELECT * FROM test");
```

Comment:
Test cases for old data format need be kept until we finished migration.

Reply:
Reverted and keep all old test cases.